lib/src/Core/T.spec.lua

```diff
@@ -46,6 +46,21 @@ return function()
 					expect(string2.typeName).to.equal("literal")
 				end
 			)
+
+			it("should convert keyOf to a union of literals", function()
+				local testTable = {
+					hey = "ready",
+					you = "set",
+					guys = 22,
+				}
+
+				local testTableKeys = T.keyOf(testTable)
+
+				expect(testTableKeys.typeName).to.equal("union")
+				expect(testTableKeys.typeParams[1].typeName).to.equal("literal")
```

Comment:
Can we test against the literal values too? Like this:

```suggestion
				expect(testTableKeys.typeParams[1].typeName).to.equal("literal")
				expect(testTableKeys.typeParams[1].typeParams[1]).to.equal("hey")
```